physionet-django/events/views.py

```diff
@@ -114,53 +116,123 @@ def event_home(request):
+                event_application.status
+                == EventApplication.EventApplicationStatus.APPROVED
+            ):
+                if EventParticipant.objects.filter(
+                    event=event, user=event_application.user
+                ).exists():
                     messages.error(request, "Application was already approved")
                     return redirect(event_home)
                 event_application.accept(
-                    comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
+                    comment_to_applicant=form.cleaned_data.get("comment_to_applicant")
                 )
                 notification.notify_participant_event_decision(
                     request=request,
                     user=event_application.user,
                     event=event_application.event,
                     decision=EventApplication.EventApplicationStatus.APPROVED.label,
-                    comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
+                    comment_to_applicant=form.cleaned_data.get("comment_to_applicant"),
                 )
-            elif event_application.status == EventApplication.EventApplicationStatus.NOT_APPROVED:
+            elif (
+                event_application.status
+                == EventApplication.EventApplicationStatus.NOT_APPROVED
+            ):
                 event_application.reject(
-                    comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
+                    comment_to_applicant=form.cleaned_data.get("comment_to_applicant")
                 )
                 notification.notify_participant_event_decision(
                     request=request,
                     user=event_application.user,
                     event=event_application.event,
                     decision=EventApplication.EventApplicationStatus.NOT_APPROVED.label,
-                    comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
+                    comment_to_applicant=form.cleaned_data.get("comment_to_applicant"),
                 )
 
             return redirect(event_home)
         else:
             form_error = True
 
+    events_all = Event.objects.all().prefetch_related(
+        "participants",
+        "applications",
+    )
+
+    event_details = {}
+    for selected_event in events_all:
+        all_applications = selected_event.applications.all()
+        pending_applications = [
+            app
+            for app in all_applications
+            if app.status == EventApplication.EventApplicationStatus.WAITLISTED
+        ]
+        rejected_applications = [
+            app
+            for app in all_applications
+            if app.status == EventApplication.EventApplicationStatus.NOT_APPROVED
+        ]
+        withdrawn_applications = [
+            app
+            for app in all_applications
+            if app.status == EventApplication.EventApplicationStatus.WITHDRAWN
+        ]
```

Comment:
Technically you could do this in one sweep instead of four, but for readability's sake I'd sake it's fine the way it is